lib/docker_client.rb

```diff
@@ -320,12 +332,11 @@ def self.image_tags
     Docker::Image.all.map { |image| image.info['RepoTags'] }.flatten.reject { |tag| tag.include?('<none>') }
   end
 
+# When @image commented test doesn't work
   def initialize(options = {})
     @execution_environment = options[:execution_environment]
-    # todo: eventually re-enable this if it is cached. But in the end, we do not need this.
-    # docker daemon got much too much load. all not 100% necessary calls to the daemon were removed.
-    #@image = self.class.find_image_by_tag(@execution_environment.docker_image)
-    #fail(Error, "Cannot find image #{@execution_environment.docker_image}!") unless @image
+    @image = self.class.find_image_by_tag(@execution_environment.docker_image)
+    fail(Error, "Cannot find image #{@execution_environment.docker_image}!") unless @image
   end
```

Comment:
Why did you re-enable that call?
To make the test pass?
Or does it have another reason?
As stated, It caused heavy load on the docker daemon, so I would suggest to switch it back off and just set the test to pending.